### PR TITLE
Fix unable to enter dates in time range

### PR DESCRIPTION
**Pull request by Quasar985**
### Prerequisites

- [x] Reviewed the [checklist](CHECKLIST.md)

- [x] Reviewed feedback from the "Sonar Cloud" bot. Note that you have to wait
    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change

Fixed issue in date time range picker, that was previously worked around by disabling the option to type in dates.

### Alternate Designs

N/A

### Why Should This Be In Core?

N/A

### Benefits

Users can now type in dates. No longer having to click many times to drastically change year.

### Possible Drawbacks

None.

### Verification Process

1. Click Arrange->Layer by time -> date range -> Absolute Range
2. Click on `From` date and try to change date by typing in text area of date picker

### Applicable Issues

https://github.com/constellation-app/constellation/issues/2142

### CODE REVIEW

**Review by Delphinus8821 — APPROVED** [review]

**Review by OrionsGuardian — CHANGES_REQUESTED** [review]
Some error handling will be required if users can freely enter any text.

Got an exception when entering an invalid date ...
image

Same error when entering a date with a different format ...
image

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java [hunk 1]
```diff
@@ -523,7 +529,15 @@ public String toString(final LocalDate date) {
 
             @Override
             public LocalDate fromString(final String s) {
-                return StringUtils.isNotBlank(s) ? LocalDate.parse(s, DATE_FORMATTER) : null;
+                if (StringUtils.isBlank(s)) {
+                    return null;
+                }
+                try {
+                    return LocalDate.parse(s, DATE_FORMATTER);
+                } catch (DateTimeException e) {
```
Comment (OrionsGuardian) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java:537]: `final`

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java [hunk 2]
```diff
@@ -523,7 +529,15 @@ public String toString(final LocalDate date) {
...
+                    LOGGER.log(Level.WARNING, "DateTimeException happened");
```
Comment (OrionsGuardian) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java:538]: Don't really need a log statement here. 
Maybe a popup error message instead ? : "Invalid date entered. Expected format: yyyy-mm-dd"

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java [hunk 3]
```diff
@@ -536,10 +550,11 @@ public LocalDate fromString(final String s) {
 
         dp.getEditor().textProperty().addListener(changed);
 
-        // The DatePicker has the horrible problem that you can type in the text field, but the value won't change,
-        // so if you type a new date and click Go, the old date will be used, not the new date that you can see.
-        // The simplest way to avoid this is to disable the text field. :-(
-        dp.getEditor().setDisable(true);
+        dp.valueProperty().addListener((v, o, n) -> {
+            if(n == null){
```
Comment (OrionsGuardian) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java:554]: formatting - spaces before/after the parenthesis

**Review by OrionsGuardian — CHANGES_REQUESTED** [review]

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java [hunk 4]
```diff
@@ -123,19 +123,21 @@ public DateTimeRangeInputPane(final PluginParameter<DateTimeRangeParameterValue>
...
+                return;
+            }
+
+            try {
+                final ZonedDateTime[] zdt = getAbsoluteRange(getZoneId());
+                isAdjusting = true;
+                if (zdt.length > 0) {
+                    parameter.setObjectValue(new DateTimeRange(zdt[0], zdt[1]));
                 }
+            } catch (DateTimeException e) {
```
Comment (antares1470) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java:136]: final

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java [hunk 5]
```diff
@@ -534,13 +552,22 @@ public LocalDate fromString(final String s) {
         final VBox picker = new VBox(dpBox, spinnerBox);
         picker.setStyle("-fx-padding:4; -fx-border-radius:4; -fx-border-color: grey;");
 
-        dp.getEditor().textProperty().addListener(changed);
+        dp.getEditor().textProperty().addListener((final ObservableValue<? extends String> v, final String o, final String n) -> {
+            if (n != null && n.isEmpty()) {
```
Comment (antares1470) [inline CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java:556]: Would it make any sense to swap this out for `StringUtils.isEmpty(n)`? Assuming you may want to do something similar if `n` is null?

**Review by antares1470 — CHANGES_REQUESTED** [review]

File: CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java (hunk 5 above)
Comment (Quasar985) [reply CorePluginFramework/src/au/gov/asd/tac/constellation/plugins/gui/DateTimeRangeInputPane.java]: I found out that `n` cant be null, so I've replaced the `n != null && n.isEmpty()` with just `StringUtils.isEmpty(n)`

**Review by antares1470 — APPROVED** [review]

**Pull request merged**